### PR TITLE
Canned TDR tables expand metadata JSON (#3067)

### PR DESCRIPTION
**Pull request by noah-aviel-dove**
Author

- [x] PR title references issue
- [x] Title of main commit references issue
- [x] PR is linked to Zenhub issue

Author (reindex)

- [x] Added `r` tag to commit title                         <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR                           <sub>or this PR does not require reindexing</sub>

Author (freebies & chains)

- [x] Freebies are blocked on this PR                       <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles              <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in the chain        <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR                <sub>or this PR is not chained to another PR</sub>

Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst  <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title                         <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR                           <sub>or this PR does not require upgrading</sub>

Author (requirements, before every review)

- [x] Ran `make requirements_update`                        <sub>or this PR leaves requirements*.txt, common.mk and Makefile untouched</sub>
- [x] Added `R` tag to commit title                         <sub>or this PR leaves requirements*.txt untouched</sub>
- [x] Added `reqs` label to PR                              <sub>or this PR leaves requirements*.txt untouched</sub>

Author (before every review)

- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>
- [x] Rebased branch on `develop`, squashed old fixups

Primary reviewer (after approval)

- [x] Commented in issue about demo expectations            <sub>or labelled issue as `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] PR title is appropriate as title of merge commit
- [x] Moved ticket to Approved column
- [x] Assigned PR to an operator

Operator (before pushing merge the commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Checked that demo expectations are clear              <sub>or issue is labeled as `no demo`</sub>
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Pushed PR branch to Github
- [x] Branch pushed to Gitlab                               <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passes in sandbox and added `sandbox` label     <sub>or PR is labeled `no sandbox`</sub>
- [x] Started reindex in `sandbox`                          <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Checked for failures in `sandbox`                     <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Added PR reference to merge commit title
- [x] Collected commit title tags in merge commit title
- [x] Moved linked issue to Merged column
- [x] Pushed merge commit to Github

Operator (after pushing the merge commit)

- [x] Moved freebies to Merged column                       <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened the PR chain                                <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate
- [ ] Pushed merge commit to Gitlab                         <sub>or this changes can be pushed later, together with another PR</sub>
- [ ] Deleted PR branch from Github and Gitlab

Operator (reindex) 

- [x] Started reindex in `dev`                              <sub>or this PR does not require reindexing or does not target `dev`</sub>
- [x] Checked for failures in `dev`                         <sub>or this PR does not require reindexing or does not target `dev`</sub>
- [x] Started reindex in `prod`                             <sub>or this PR does not require reindexing or does not target `prod`</sub>
- [x] Checked for failures in `prod`                        <sub>or this PR does not require reindexing or does not target `prod`</sub>

Operator

- [ ] Unassigned PR

### CODE REVIEW

File: test/indexer/data/1b6d8348-d6e9-406a-aa6a-7ee886e52bf9.tables.tdr.json [hunk 1]
```diff
@@ -23,7 +100,52 @@
...
+                            ]
+                        },
+                        "type": "primary",
+                        "model_organ": {
+                            "text": "immune system",
+                            "ontology": "UBERON:0002405",
+                            "ontology_label": "immune system"
+                        },
+                        "growth_conditions": {
+                            "growth_medium": "Dulbecco\u2019s modified Eagle\u2019s medium supplemented with 10% (v/v) FBS, 2 mM l-glutamine, and antibiotics (penicillin and streptomycin, 10 \uf06dg/ml)",
```
Comment (amarjandu) [inline test/indexer/data/1b6d8348-d6e9-406a-aa6a-7ee886e52bf9.tables.tdr.json:121]: here

**Review by amarjandu — APPROVED** [review]
LGTM.
Not related to this PR but it looks like there are some characters are not decoded correctly in the canned bundle.

`\u2019`

File: test/indexer/data/1b6d8348-d6e9-406a-aa6a-7ee886e52bf9.tables.tdr.json (hunk 1 above)
Comment (noah-aviel-dove) [reply test/indexer/data/1b6d8348-d6e9-406a-aa6a-7ee886e52bf9.tables.tdr.json:121]: Good catch, we should triage this

File: test/indexer/test_tdr.py [hunk 2]
```diff
@@ -168,15 +169,24 @@ def _make_mock_entity_table(self,
                                 rows: JSONs) -> None:
         schema = self._bq_schema(rows[0])
         columns = {column['name'] for column in schema}
-        # TinyQuery's errors are typically not helpful in debugging missing/
-        # extra columns in the row JSON.
-        for row in rows:
+        serialized_columns = {'content', *(('descriptor',) if table_name.endswith('_file') else ())}
```
Comment (hannes-ucsc) [inline test/indexer/test_tdr.py:172]: line too long

Comment (hannes-ucsc) [reply test/indexer/test_tdr.py:172]: can't you do this dynamically by just looking at the value being a mapping?

Comment (hannes-ucsc) [reply test/indexer/test_tdr.py:172]:
```suggestion
        json_columns = {'content', *(('descriptor',) if table_name.endswith('_file') else ())}
```

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

File: test/indexer/data/1b6d8348-d6e9-406a-aa6a-7ee886e52bf9.tables.tdr.json (hunk 1 above)
Comment (hannes-ucsc) [reply test/indexer/data/1b6d8348-d6e9-406a-aa6a-7ee886e52bf9.tables.tdr.json:121]: What do you mean by that? Discuss in PL?

Comment (noah-aviel-dove) [reply test/indexer/data/1b6d8348-d6e9-406a-aa6a-7ee886e52bf9.tables.tdr.json:121]: I was going to open a new ticket but I changed my mind and now I want to discuss during PL.

File: test/indexer/test_tdr.py [hunk 3]
```diff
@@ -168,15 +172,23 @@ def _make_mock_entity_table(self,
...
-        for row in rows:
+
+        def dump_row(row: JSON) -> str:
             row_columns = row.keys()
+            # TinyQuery's errors are typically not helpful in debugging missing/
+            # extra columns in the row JSON.
             assert row_columns == columns, row_columns
+            row = {
+                column_name: (json.dumps(column_value)
+                              if isinstance(column_value, Mapping) else column_value)
```
Comment (hannes-ucsc) [inline test/indexer/test_tdr.py:183]:
```suggestion
                              if isinstance(column_value, Mapping) else 
                              column_value)
```

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

**Review by hannes-ucsc — APPROVED** [review]

**Pull request merged**